repo/csvRepo.go

```diff
@@ -0,0 +1,103 @@
+package repo
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/GabrielRendonP/ondemand-go-bootcamp/entities"
+)
+
+type localData struct{}
+
+type LocalDataInterface interface {
+	ReadCSVData() ([][]string, error)
+	GetAllPokemonsApi() []entities.Pokemon
+	SaveToCsv([]entities.Pokemon) error
+}
+
+func NewLocalData() LocalDataInterface {
+	return &localData{}
+}
+
+// Probably should go in a driver
+func (r localData) ReadCSVData() ([][]string, error) {
+	var data [][]string
+
+	file, err := os.Open("./lib/pokemon.csv")
+	if err != nil {
+		fmt.Println("is this error")
+		return nil, err
+	}
+
+	defer file.Close()
+
+	csvReader := csv.NewReader(file)
+	data, err = csvReader.ReadAll()
+
+	if err != nil {
+		log.Fatal(err)
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func (r localData) GetAllPokemonsApi() []entities.Pokemon {
```

Comment:
This should not be here, remember we want to have our layers separated, this means CSV manipulation should be on a different layer rather than mixed.